lib/pure/math.nim

```diff
@@ -1211,6 +1211,32 @@ func `^`*[T: SomeNumber](x: T, y: Natural): T =
         break
       x *= x
 
+func isInteger(y: SomeFloat): bool =
+  ## Determines if a float represents an integer
+  ## Note this might trigger a change of the rounding mode.
+  # In C++, we usually use the `rint` function.
+  return round(y) == y
+
+func `^`*[T: SomeNumber, U: SomeFloat](x: T, y: U): float =
+  ## Computes `x` to the power of `y`.
+  ##
+  ## Error handling follows C++ specification
+  ## https://en.cppreference.com/w/cpp/numeric/math/pow
+  let
+    isZero_x: bool = (x == 0.0 or x == -0.0)
+    isNegZero: bool = classify(x) == fcNegZero
+    isPosZero: bool = classify(x) == fcZero
+    # y_isOddInteger: bool = (isInteger(y) and (int(y) mod 2 == 1))
+    y_isFinite: bool = (y != Inf and y != -Inf)
```

Comment:
According to the style guide https://nim-lang.github.io/Nim/nep1.html#style-guidelines-naming-conventions canekCase is preferred

Reply:
Fixed.